echopype/echodata/echodata.py

```diff
@@ -312,6 +312,145 @@ def squeeze_non_scalar(n):
+
+        Parameters
+        ----------
+        extra_platform_data : xr.Dataset
+            An `xr.Dataset` containing the additional platform data to be added
+            to the `EchoData.platform` group.
+        time_dim: str, default="time"
+            The name of the time dimension in `extra_platform_data`; used for extracting
+            data from `extra_platform_data`.
+
+        Examples
+        --------
+        >>> ed = echopype.open_raw(raw_file, "EK60")
+        >>> extra_platform_data = xr.open_dataset(extra_platform_data_file)
+        >>> ed.update_platform(extra_platform_data)
+        """
+
+        # # only take data during ping times
+        # start_time, end_time = min(self.beam["ping_time"]), max(self.beam["ping_time"])
+
+        # Handle data stored as a CF Trajectory Discrete Sampling Geometry
+        # http://cfconventions.org/Data/cf-conventions/cf-conventions-1.8/cf-conventions.html#trajectory-data
+        # The Saildrone sample data file follows this convention
+        if (
+            "featureType" in extra_platform_data.attrs
+            and extra_platform_data.attrs["featureType"].lower() == "trajectory"
+        ):
+            for coordvar in extra_platform_data.coords:
+                if (
+                    "cf_role" in extra_platform_data[coordvar].attrs
+                    and extra_platform_data[coordvar].attrs["cf_role"]
+                    == "trajectory_id"
+                ):
+                    trajectory_var = coordvar
+
+            # assumes there's only one trajectory in the dataset (index 0)
+            extra_platform_data = extra_platform_data.sel(
+                {trajectory_var: extra_platform_data[trajectory_var][0]}
+            )
+            extra_platform_data = extra_platform_data.drop_vars(trajectory_var)
+            extra_platform_data = extra_platform_data.swap_dims({"obs": time_dim})
+
+        platform = self.platform.assign_coords(
+            time2=extra_platform_data[time_dim].values
+        )
+        platform["time2"].attrs[
+            "long_name"
+        ] = "time dimension for external platform data"
+
+        dropped_vars = []
+        for var in ["pitch", "roll", "heave", "latitude", "longitude", "water_level"]:
+            if var in platform and platform[var].isnull().all():
+                dropped_vars.append(var)
+        if len(dropped_vars) > 0:
+            warnings.warn(
+                f"some variables will be overwritten by platform data: {', '.join(dropped_vars)}"
+            )
+        platform = platform.drop_vars(
+            ["pitch", "roll", "heave", "latitude", "longitude", "water_level"],
+            errors="ignore",
+        )
+
+        num_obs = len(extra_platform_data[time_dim])
+
+        def mapping_get_multiple(mapping, keys, default=None):
+            for key in keys:
+                if key in mapping:
+                    return mapping[key].data
+            return default
```

Comment:
@imranmaj: this function would only return the first existing keys in the mapping data set, right?

Seems that this could potentially produce bugs down the line, because the input argument is call `keys`, so users may think that a list with each element being an array of values corresponding to **all** of what's listed in `keys` would be generated, while in fact only the first existing key will be returned.

The current behavior of this function does exactly what you need this function to do in the code below, but in that case, the function name should not be `mapping_get_multiple`, because it would only get data from a single variable.

Which way do you intend this to be? A function that only returns data from 1 variable from all of the possible names, or a function that would return data from multiple variables?

Reply:
> users may think that a list with each element being an array of values corresponding to all of what's listed in keys would be generated, while in fact only the first existing key will be returned.

It should only be used internally, I don't believe users can use it.

> Which way do you intend this to be? A function that only returns data from 1 variable from all of the possible names, or a function that would return data from multiple variables?

It should return data from 1 variable, it's "searching" for multiple possible names to be flexible.

> the function name should not be `mapping_get_multiple`, because it would only get data from a single variable.

How about `mapping_try_get_multiple`?

Reply:
> It should only be used internally, I don't believe users can use it.

I should rephrase this to say that it is not obvious to _others_.

> How about mapping_try_get_multiple?

I think the "multiple" part could be confused with multiple variables. How about `mapping_search_variable`?